### PR TITLE
Add filters for FD, e.g. KO dissipation

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/NumericalAlgorithms/FiniteDifference/Filter.cpp [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+          filtered_data, buffer, volume_vars, ghost_cell_vars, volume_mesh,
+          number_of_variables, args...);
+      break;
+    case 10:
+      ::fd::filter_impl<FilterComputer<10, true>>(
+          filtered_data, buffer, volume_vars, ghost_cell_vars, volume_mesh,
+          number_of_variables, args...);
+      break;
+    default:
+      ERROR("Cannot do finite difference derivative of order " << fd_order);
```
Comment (nilsdeppe) [inline src/NumericalAlgorithms/FiniteDifference/Filter.cpp]: Change wording of error

File: tests/Unit/NumericalAlgorithms/FiniteDifference/Test_Filter.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          mesh.number_of_grid_points() * number_of_vars, 0.0);
+      auto filtered_vars_span =
+          gsl::make_span(filtered_vars.data(), filtered_vars.size());
+
+      fd::low_pass_filter(
+          make_not_null(&filtered_vars_span),
+          gsl::make_span(volume_vars.data(), volume_vars.size()),
+          ghost_cell_vars, mesh, number_of_vars, deriv_order, epsilon);
+
+      // Get only the KO dissipation term by subtracting off the volume
```
Comment (kidder) [inline tests/Unit/NumericalAlgorithms/FiniteDifference/Test_Filter.cpp]: this is not KO so I am confused by the comment

File: src/NumericalAlgorithms/FiniteDifference/Filter.hpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+ * derivative so as not to spoil the accuracy of the scheme. That is, for
+ * second order FD, \f$m=2\f$ and one should use \f$\mathcal{D}^{(4)}\f$.
+ * However, this choice requires a larger stencil and whether or not this is
+ * necessary also depends on when and how in the algorithm the operators are
+ * applied.
+ *
+ * We arrive at the following operators:
+ *
+ * \f{align*}{
+ *  \mathcal{D}^{(2)}f_i&=-\frac{\epsilon}{\Delta x}\epsilon
```
Comment (kidder) [inline src/NumericalAlgorithms/FiniteDifference/Filter.hpp]: extra epsilon

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
Rebased, fixed, and squashed. Thanks for the review :D

**Review by kidder — APPROVED** [review]

**Pull request merged**